BoxBrowseSDK/BoxBrowseSDK/BOXFolderViewController.m

```diff
@@ -400,44 +400,65 @@ - (void)fetchItemsWithCompletion:(void (^)(NSArray *items))completion
     folderRequest.SDKIdentifier = BOX_BROWSE_SDK_IDENTIFIER;
     folderRequest.SDKVersion = BOX_BROWSE_SDK_VERSION;
     folderRequest.requestAllFolderFields = YES;
-    [folderRequest performRequestWithCompletion:^(BOXFolder *folder, NSError *error) {
-        if (error) {
-            [self didFailToLoadItemsWithError:error];
-            self.navigationController.toolbarHidden = YES;
+
+    void (^completionBlock)(NSArray *, BOOL, NSError *) = ^void(NSArray *items, BOOL fromCache, NSError *error) {
+        dispatch_async(dispatch_get_main_queue(), ^{
+            if (items.count > 0) {
+                [self switchToNonEmptyView];
+            } else if (!fromCache && self.tableView.visibleCells.count < 1) {
+                [self switchToEmptyStateWithError:error];
+            }
+
+            if (completion) {
+                completion(items, fromCache, error);
+            }
+        });
+    };
+
+    void (^itemFetchBlock)(BOXFolder *, BOOL fromCache, NSError *) = ^void(BOXFolder *folder, BOOL fromCache, NSError *error) {
+        if (folder && !error) {
+            [self setCurrentFolder:folder];
+        }
+
+        if (error.code == BOXContentSDKAPIErrorBadRequest ||
+            error.code == BOXContentSDKAPIErrorUnauthorized ||
+            error.code == BOXContentSDKAPIErrorForbidden ||
+            error.code == BOXContentSDKAPIErrorNotFound) {
+            completionBlock(nil, fromCache, error);
```

Comment:
The UI displays an empty folder with a message along the lines of "unable to load this folder." I prefer this route over switchToErrorStateWithError since that method creates a HUD that displays the message temporarily and then disappears, leaving the user with a blank screen. This change  resembles the current behavior in bender.